standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -584,6 +584,32 @@ public OpenTxnsResponse openTxns(OpenTxnRequest rqst) throws MetaException {
           LOG.debug("Going to execute update <" + q + ">");
           stmt.execute(q);
         }
+
+        if (rqst.isSetReplPolicy()) {
+          List<String> rowsRepl = new ArrayList<>();
+          List<String> selectRepl = new ArrayList<>();
+
+          for (int i = 0; i < numTxns; i++) {
+            selectRepl.add("select TM_TARGET_TXN_ID from REPL_TXN_MAP where TM_REPL_POLICY = " + quoteString(
```

Comment:
Need to verify if txn is there in REPL_TXN_MAP before allocating it at target. Here, we leak txns at target if we apply the same event twice.

Logic should be as follows.
1. Check if any txn exist in REPL_TXN_MAP, then reduce the numTxns to allocate. Or if we want to strictly assume no duplication open on same txn, then shall throw exception.
2. If txn repeat in the list also need to reduce the numTxns to allocate.
3. Allocate txns only for remaining ones and add entries to REPL_TXN_MAP.

Reply:
1. Can not throw exception as the method is idempotent.
2. Need to skip txns already present but that is not possible ..either all transactions are present or nothing should be present. may be an assert should suffice.

Reply:
need to add the event id range check in replTxn task

Reply:
done